enhancements/builds/disable-jenkins-pipeline.md

```diff
@@ -0,0 +1,257 @@
+* Document how JenkinsPipeline strategy builds can be migrated off of OpenShift.
+
+### Non-Goals
+
+* Migrate Jenkins users to OpenShift Pipelines/Tekton.
+* Stop distributing the Jenkins image and our associated OpenShift plugins.
+
+## Proposal
+
+### User Stories
+
+As a developer using Jenkins on OpenShift
+I want to continue using my Jenkinsfiles
+So that I can continue to use Jenkins for my CI/CD processes
+
+As an OpenShift release engineer
+I want to remove Jenkins from the OCP payload
+So that I can reduce the size of the payload
+And release Jenkins fixes on their own cadence
+
+### Implementation Details/Notes/Constraints [optional]
+
+Deprecation warnings for the Jenkins Pipeline Build Strategy will need to be updated in the next OpenShift release (OCP 4.N).
+Since Jenkins integrations have been officially been officially deprecated for well over one year, these warnings need to be strengthened so customers are fully aware that all direct Jenkins integrations will be removed.
+Like the use of other deprecated APIs, OpenShift must also alert users if a Jenkins pipeline build strategy is used in a Build or BuildConfig.
+
+In the OCP 4.(N+1) release, builds which use the Jenkins pipeline build strategy should fail immediately.
+All documentation related to Jenkins pipeline builds will need to be removed, and replaced with a statement that the Jenkins pipeline strategy has been disabled.
+The updated documenation should include instructions on how to move a Jenkinsfile off of OpenShift.
+
+Ideally, the deprecation warnings are strengthened in OCP 4.9.
+Jenkins pipeline builds will then be disabled in OCP 4.10 (our next Extended User Support release).
+
+#### Jenkins Pipeline Build Strategy
+
+In OCP 4.N, an alert should be fired if a build is started with the Jenkins Pipeline build strategy.
+Likewise, an alert should be fired if there is any BuildConfig which uses the Jenkins Pipeline strategy.
+This data is already collected by openshift-state-metrics, therefore creating an alert is a matter of aggregating the right metrics.
+
+In OCP 4.(N+1), the build controller should immediately fail a build which utilizes the Jenkins pipeline strategy with an appropriate failure message.
+
+#### oc <new|start>-build
+
+In OCP 4.N, `oc new-build` should print a warning message if the generated/referenced BuildConfig uses the Jenkins pipeline strategy.
+`oc start-build` should print a warning message if a build was started with the Jenkins pipeline strategy.
+
+In OCP 4.(N+1), `oc new-build` should not support the creation of new Builds with the Jenkins pipeline strategy.
+`oc start-build` may continue to create builds referencing the Jenkins pipeline strategy, but these builds will ultimately fail.
+
+#### Web Console
+
+User will continue to be able to deploy Jenkins on OpenShift by instantiating our provided Templates.
+
+In OCP 4.N, the Jenkins pipeline view of a build needs to display a warning that the strategy will be removed.
```

Comment:
Similarly to oc, this should happen asap, if this functionality is already deprecated.